hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -3719,9 +3728,12 @@ private Expression visitTableGroup(TableGroup tableGroup, SqmFrom<?, ?> path) {
 			if ( inferredEntityMapping == null ) {
 				// When the inferred mapping is null, we try to resolve to the FK by default, which is fine because
 				// expansion to all target columns for select and group by clauses is handled in EntityValuedPathInterpretation
-				if ( entityValuedModelPart instanceof EntityAssociationMapping && ( (EntityAssociationMapping) entityValuedModelPart ).isFkOptimizationAllowed() ) {
+				if ( entityValuedModelPart instanceof EntityAssociationMapping
+						&& ( (EntityAssociationMapping) entityValuedModelPart ).isFkOptimizationAllowed()
+						&& !isLeftJoinWithCondition( path ) ) {
```

Comment:
Actually, it doesn't matter what the join type is. Even for `inner` it is problematic to use the FK if there are additional join conditions.

Reply:
You're right, thanks for the review - updated the PR.